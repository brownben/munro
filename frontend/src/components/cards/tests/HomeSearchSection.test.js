import { mount } from '@vue/test-utils'
import HomeSearchSection from '@/components/cards/HomeSearchSection.vue'

test('Is a Vue Instance', () => {
  const wrapper = mount(HomeSearchSection, {
    stubs: ['router-link'],
    mocks: {
      $router: { push: jest.fn() },
    },
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(HomeSearchSection, {
    stubs: ['router-link'],
    mocks: {
      $router: { push: jest.fn() },
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})
