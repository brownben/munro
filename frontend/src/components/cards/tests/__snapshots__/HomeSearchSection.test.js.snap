// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<section
  class="w-full py-10 text-white bg-main-500"
>
  <div
    class="flex flex-col items-center max-w-screen-xl px-6 mx-auto text-center sm:px-12 lg:px-8"
  >
    <h2
      class="mb-2 text-4xl leading-tight tracking-tighter md:text-5xl font-heading"
    >
      
      Find 
      <span
        class="font-bold tracking-normal"
      >
        Everything
      </span>
       You Need
    
    </h2>
     
    <h3
      class="text-lg font-light md:text-xl font-heading"
    >
      
      Instantly Search All Leagues, Events and Competitors
    
    </h3>
     
    <form
      class="relative block w-full mt-6 md:mt-8 sm:w-3/4 lg:w-1/2 xl:w-1/3"
    >
      <label
        class="sr-only"
        for="searchBox"
      >
        Search Query:
      </label>
       
      <input
        class="w-full py-2 pl-10 pr-24 font-sans leading-snug transition duration-300 ease-in-out bg-transparent border border-white border-opacity-50 outline-none appearance-none text-main-100 focus:border-opacity-100 focus:text-white rounded-shape"
        id="searchBox"
        spellcheck="false"
        type="text"
      />
       
      <button
        class="absolute top-0 right-0 px-4 py-2 transition duration-300 outline-none font-heading rounded-shape bg-main-100 text-main-600 hover:bg-white focus:bg-white"
      >
        
        Search
      
      </button>
       
      <svg
        class="absolute top-0 left-0 w-6 h-6 py-px my-2 ml-2 text-main-100"
        fill="currentColor"
        viewBox="0 0 20 20"
      >
        <path
          clip-rule="evenodd"
          d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
          fill-rule="evenodd"
        />
      </svg>
    </form>
  </div>
</section>
`;
